src/history/hs_rec.c

```diff
@@ -407,6 +407,14 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_MULTI *mult
         enable_reverse_modify =
           (WT_STREQ(btree->value_format, "S") || WT_STREQ(btree->value_format, "u"));
 
+        /*
+         * If there exists an on page tombstone without a timestamp, consider it as a no
+         * timestamp update to clear the timestamps of all the updates that are inserted into the
+         * history store.
+         */
+        if (list->onpage_tombstone != NULL && list->onpage_tombstone->start_ts == WT_TS_NONE)
```

Comment:
Can we move this to the reconciliation code instead of doing it here?

Reply:
I didn't see any way to fix this problem in the reconciliation flow as the updates in the update list have the timestamps and they cannot be modified. Whereas in this flow, we can control the time window that gets inserted into the history store.

Reply:
true